# Python 2024 onlineSummer internship

# Make your place in Top 20% python programmers of the world

## **Name:Usfa Sohail**
## **E-mail:usfasohail389@gmail.com**

# **To Do Task 1:**

## **Write prompts to fulfill following needs**

## **Information 1:**

## **Step by step process to write a python program**

### **1.Set Up Your Development Environment**

**Install Python:** Ensure Python is installed on your system. Download it from the official Python website.

**Choose a Code Editor or IDE:** Options include VSCode, PyCharm, or Jupyter Notebook. Install your preferred editor or IDE.

### **2.Create a New Python File**

Open your code editor or IDE.

Create a new file with a .py extension (e.g., hello.py).

### **3.Understand the Basics of Python Syntax**

**Comments:** Learn how to write single-line and multi-line comments.

**Indentation:** Python uses indentation to define code blocks. Be consistent with the number of spaces you use for indentation.

### **4.Write a Simple Python Program**

Start by writing a simple program, like printing "Hello, world!" to the console.


### **5.Learn About Variables and Data Types**

Understand how to declare variables and the different data types available in Python (e.g., strings, integers, floats, booleans).


### **6.Perform Basic Operations**

Practice arithmetic operations, string concatenation, and other basic operations.
### **7.Use Control Structures**

Learn how to write conditional statements (if, elif, else) and loops (for, while) to control the flow of your program.


### **8.Define and Use Functions**

Understand how to define functions and call them. Functions help you organize your code into reusable blocks.
### **9.Work with Modules and Libraries**

Learn how to import and use Python's standard libraries and third-party libraries to extend the functionality of your programs.

### **10.Handle Errors and Exceptions**

Learn about error handling using try, except, and finally blocks to manage exceptions in your code.
### **11.Read from and Write to Files**

Practice file I/O operations to read data from files and write data to files.
### **12.Test Your Code**

Regularly run your Python program to check for errors and ensure it behaves as expected.

### **13.Debugging**

Use debugging tools or insert print statements in your code to find and fix errors.
### **14.Practice and Build Projects**

Apply your knowledge by building small projects, which will help reinforce what you've learned and improve your problem-solving skills.

## **Infromation 2:**

## **Step by step process to write a python comment**

 ### **1.Understand the Purpose of Comments:**
Comments help explain what your code does and why. They make your code easier to read and maintain.

### **2. Decide on the Type of Comment:**
##### **There are two main types:**

**Inline Comments:** Short comments next to a line of code.

**Block Comments:** Longer comments that cover several lines or a section of code.
### **3. Write Inline Comments:**
**When:** Use these for brief explanations right beside a line of code.

**How:** Place the comment after the code, with a few spaces between the code and the comment.
### **4. Write Block Comments:**
**When:** Use these for longer explanations or for describing a section of code.

**How:** Put the comment above the code you are explaining, with each line of the comment starting with a #.
### **5. Write Docstrings for Functions and Classes:**
**When:** Use these to describe what a function or class does.

**How:** Write the comment right after the function or class definition, using triple quotes.
### **6. Update Comments:**
**Why:** Ensure comments stay accurate as you change the code.

**How:** Revise or remove comments if the code changes.
### **7. Be Clear and Brief:**
**What:** Focus on making your comments clear and to the point.

**How:** Avoid redundant explanations and keep your comments focused.
### **8. Explain “Why” Not “What”:**
**What:** Aim to clarify why something is done a certain way.

**How:** The code should show what it does; your comments should explain the reasoning.

# Your Turn 1:

## Step by step process to debug a python program
### **1.Understand the Problem:**

Clearly define the nature of the problem in your code.

**Reproduce the Issue:** Run the program to see the error in action.

**Document Symptoms:** Note down what goes wrong. Is there a specific error message, or is the output not as expected?

**Define Expectations:** Compare what you expected to happen with what actually happened.
### **2.Analyze the Error Message:**

Extract useful information from the error message and traceback.

**Read Traceback:** Trace the error back to the exact line where it occurred.

**Identify Error Type:** Determine whether it’s a syntax error, runtime error, or logical error.

**Research Error:** Look up the error message or exception type for more information if it’s unfamiliar.
### **3.Add Print Statements:**

Gather information about the internal state of the program.

**Insert Prints:** Add statements to display variable values or control flow.

**Identify Checkpoints:** Place them at key points before and after suspected problematic code.

**Observe Outputs:** Use the outputs to check if variables hold expected values and if functions execute as intended.
### **4.Use a Debugger:**

**Set Breakpoints:** Pause at the line where the addition occurs.

**Step Through:** Examine the variables and their types.

### 5.**Check for Common Issues:**

**Verify:** Make sure you’re not accidentally mixing data types.

**Dependencies:** Ensure no external libraries are causing type conflicts.

### **6.Review Code Logic:**

**Analyze:** Check if the addition operation is supposed to be concatenation or arithmetic.

**Edge Cases:** Test with different data types.

### **7.Isolate the Problem:**

**Simplify:** Create a minimal example to isolate the TypeError.

### **8.Consult Documentation and Resources:**

**Read:** Look up type conversion and data manipulation in Python.

**Search:** Look for TypeError solutions for similar issues.
### **9.Refactor the Code:**

**Fix:** Convert the integer to a string or use arithmetic operations correctly.

**Re-test:** Ensure that the fix works and does not break other functionality.
### **10.Test After Fixing:**

**Run Tests:** Check the program to confirm the fix and that no new issues are introduced.


# To Do Task 2:

### Example 1:Print the following message on output screen:
Purpose of our lives is:Friendship of Allah

In [3]:
print("Purpose of our lives is:Friendship of Allah")

Purpose of our lives is:Friendship of Allah


### Example 2:print the following information on screen:
Your Name

Your age

Your marks in Fsc(in persentage)

In [1]:
print("Usfa Sohail")
print("19 years")
print("89%")

Usfa Sohail
19 years
89%


### Example 3:print the following information about your city:
Name

Population

Famous for 

Geographic area in kilometers

In [5]:
print("Lahore")
print("14.4 million people")
print("Cultural Heritage")
print("1772 sq kilometers")

Lahore
14.4 million people
Cultural Heritage
1772 sq kilometers


# Your Turn:
### Example 1:print the names of all subjects in first semester

In [2]:
print("Functional English")
print("Programming fundamentals")
print("AICT")
print("Islamiat")
print("Physics")

Functional English
Programming fundamentals
AICT
Islamiat
Physics


### Example 2:print the provinces of pakistan

In [3]:
print("Punjab")
print("Sindh")
print("KPK")
print("Balochistan")

Punjab
Sindh
KPK
Balochistan


### Example 3:print 4 fruits name

In [8]:
print("Apple,Mango,Banana,Orange")

Apple,Mango,Banana,Orange


## To Do Task 3:
### Example 1:Use single line and multi line comment in your program for the following attributes
### Developer name 
### Developer email id
### python IDE
### Program name
### program version
### start date
### end date

In [2]:
print("Rao Muhammad Adeel Nawab") #Developer name
print("abcg786@gmail.com") #Developer email id
print("jupyter Notebook") #python IDE
print("Use of comments") #program name
print("python 3") #program using version
print("9/7/2024") #start date
print("9/7/2024") #end date

Rao Muhammad Adeel Nawab
abcg786@gmail.com
jupyter Notebook
Use of comments
python 3
9/7/2024
9/7/2024


## Your Turn:
### Three programs with single and multi line comments
### Example 1:

In [3]:
a=12
b=4
c=a+b #adding two numbers and assigning value to c
print("Value after adding both numbers is:",c) #displaying the value of c

Value after adding both numbers is: 16


### Example 2:

In [5]:
name="Usfa Sohail" #assigning string value to name variable
age=19 #assigning value to age variable
print("My name is",name) #printing name 
print("My age  is",age) #printing age

My name is Usfa Sohail
My age  is 19


### Example 3:

In [6]:
""" This program multiplies two numbers with each other.
It also stores the result of both the numbers and then display its result."""
num1=4
num2=6
product=num1*num2 #product obtaining by multiplying two numbers and storing it in variable name product
print("Product of two numbers is:",product)

Product of two numbers is: 24


## To Do Task 3:
### Example 1: Create variables to store your demographic information like:
### Name
### Gender
### age
### intrest
### Hobby
### Address

In [7]:
name="Usfa SOhail"
gender="Female"
age=19
intrest="coding"
hobby="Reading newspaper"
address="389 A2 township Lahore"
print("Name:",name)
print("Gender:",gender)
print("age:",age)
print("Mine intrest:",intrest)
print("My hobby is:",hobby)
print("My address:",address)

Name: Usfa SOhail
Gender: Female
age: 19
Mine intrest: coding
My hobby is: Reading newspaper
My address: 389 A2 township Lahore


### Your Turn:

In [8]:
a=3.4
b=19
c="apple"
print(type(a))
print(type(b))
print(type(c))

<class 'float'>
<class 'int'>
<class 'str'>


## To Do Task 4:
### Example:write result of the folllowing expressions
### 1) 15//3%2+5-2*5
### 2) 15//(3%2)+(5-2)*5
### 3) 15/3%2+5-2**3

In [2]:
result1=15//3%2+5-2*5
result2=15//(3%2)+(5-2)*5
result3=15/3%2+5-2**3
print("Result of first equation=",result1)
print("Result of second equation=",result2)
print("Result of third equation=",result3)

Result of first equation= -4
Result of second equation= 30
Result of third equation= -2.0


## Your Turn:
### Write down different arithmetic equation and write down there results

In [5]:
result1=(10+20)//5%3*2-7
result2=25/(4%3+1)-(3+2)*4
result3=18//3%4+6-6
print("Result of first equation=",result1)
print("Result of second equation=",result2)
print("Result of third equation=",result3)

Result of first equation= -7
Result of second equation= -7.5
Result of third equation= 2


## To Do Task 5:
### Write pseudocode of following programs:
### 1) Multiplying three numbers
### 2) finding area of a circle
### 3) finding square of a number

## Pseudocode of multiplying three numbers:

## Pseudocode of finding area of circle

## Pseudocode to find square of a number

## Your Turn:
# Pseudocode to find cube of a number

## TO DO Task:
## Using template based approch
## Example 1:Multiplication of three numbers

### Step 1:real word scenario:
program for multiplication of three numbers
### Step 2:client requirement
simple program that multiplies three integer type numbers
### Step 3:input-processing-output
### input:
No of inputs=3
Data types:integer
### instruction:
multiply
### processing:
number1 * number2 * number3
### Output:
No of output:1
Data type:integer

### Step 4: Pseudocode

### Step 5:Flowchart drawn

### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:multiplication of three numbers

### Step 8+9:Convert pseudocode in code and test that code
### Data for testing the code:
number1=5

number2=7

number3=2

In [9]:
#input
number1=int(input("Enter the first number:")) #Taking first number as an int
number2=int(input("Enter the second number:")) #Taking second number as an int
number3=int(input("Enter the third number:")) #Taking third number as an int 
product=number1*number2*number3 #Calculating product
print("The product=",product) #displaying product of three numbers

Enter the first number: 5
Enter the second number: 7
Enter the third number: 2


The product= 70


### Step 10: Output of code:
The product=70

## Example 2:Calculate square of a number

### Step 1:real word scenario:
program for calculating square of a number
### Step 2:client requirement
simple program that find square of a integer type number
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:integer
### instruction:
multiply
### processing:
number*number
### Output:
No of output:1
Data type:integer

### Step 4: Pseudocode

### Step 5:Flowchart drawn

### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find square of a number

### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5

In [2]:
#input
number=int(input("Enter number:")) #Taking int number as a input from user
square=number*number
print("The square=",square)

Enter number: 5


The square= 25


### Step 10: Output of code:
The square=25

## Example 3:Find area of a circle
### Step 1:real word scenario:
program for finding area of a circle
### Step 2:client requirement
simple program that finds area of a circle
### Step 3:input-processing-output
### input:
No of inputs=1

Data types:float
### instruction:
multiply
### processing:
pi* radius* radius
### Output:
No of output:1

Data type:float

### Step 4:Pseudocode

### Step 5: Flow chart drawn

### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find area of a circle

### Step 8+9:Pseudocode inton  code and testing of code
### Data for testing of code:
radius=7.8

In [11]:
#input
radius=float(input("Enter radius:")) #Taking radius as an input of float datatype
pi=3.14 #setting pi to 3.14
area=pi*radius*radius #calulating area
print("The area=",area) #printing area of a circle

Enter radius: 7.8


The area= 191.0376


## Step 10:Output of code
The area=191.0376

## Example 4:Print a two digit number in reverse order
### Step 1:real word scenario:
program to print a two digit number in reverse order
### Step 2:client requirement
simple program that prints the two two digit number in reverse order
### Step 3:input-processing-output
### input:
No of inputs=1

Data types:int
### instruction:
division,mod
### processing:
first_digit=number//10

second_digit=number%10
### Output:
No of output:1

Data type:int

### Step 4:Pseudocode

### Step 5: Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To print two digit number in reverse order

### Step 8+9:Pseudocode into code and testing of that code
### Data for testing of code:
number=25

In [15]:
#input
number=int(input("Enter two digit number:")) #Taking number as an int datatype
first_digit=number//10 #Calulating first digit from number
second_digit=number%10 #calculating second digit from number
print((second_digit*10)+first_digit)

Enter two digit number: 25


52


### Step 10:Output of code:
52

## Example 5:printing three digit number in reverse order
### Step 1:real word scenario:
program to print a three digit number in reverse order
### Step 2:client requirement
simple program that prints the three digit number in reverse order
### Step 3:input-processing-output
### input:
No of inputs=1

Data types:int
### instruction:
division,mod
### processing:
first_digit=number//100

second_digit=(number//10)%10

third_digit=number%10

### Output:
No of output:1

Data type:int

### Step 4:Pseudocode

### Step 5:Flowchart drawn

### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To print three digit number in reverse order

### Step 8+9:Pseudocode into code and testing of code
### Data for testing of code
number=678

In [18]:
#input
number=int(input("Enter three digit number:")) #Taking input three digit number as an int datatype
first_digit=number//100 #calculating first digit
second_digit=(number//10)%10 #calculating second digit
third_digit=number%10 #calculating third digit
#output
print((third_digit*100)+(second_digit*10)+first_digit) #displaying number in reverse order

Enter three digit number: 678


876


### Step 10: Output of a code:
876

## Your Turn:
## Find cube of a number

### Step 1:real word scenario:
program for calculating cube of a number
### Step 2:client requirement
simple program that find cube of a integer type number
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:integer
### instruction:
multiply
### processing:
number * number * number
### Output:
No of output:1
Data type:integer
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find cube of a number
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5

In [1]:
#input
number=int(input("Enter number:")) #Taking int number as a input from user
cube=number*number*number
print("The cube=",cube)

Enter number: 5


The cube= 125


### Step 10: Output of code:
The square=25

# TO DO Task:
## Take input from user as an integer and tell its negative or positive number or zero
### Step 1:real word scenario:
program to find that number is negative positive or zero
### Step 2:client requirement
simple program that finds that number is negative positive or zero
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:integer
### instruction:
comparision
### processing:
num>0
num<0
num==0
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find that number is neg pos or zero
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5

In [1]:
#input
num=int(input("Enter the value:"))#taking input as int and storing in num variable
if num==0:#comparing the number with zero
    print("Number is zero")#printing prompt
if num<0:#comparing number
    print("Number is negative")#printing prompt
if num>0:#comparing number
    print("Number is positive")#printing prompt

Enter the value: 5


Number is positive


### Output of program
Number is positive

# TO DO Task:
## Find that person is elegible for vote or not
### Step 1:real word scenario:
program to find that person is eligible for vote or not
### Step 2:client requirement
simple program that finds that person is eligible for vote or not
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:integer
### instruction:
comparision
### processing:
age>18
age<18
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find that person is elegible for vote or not
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=19

In [2]:
#input
age=int(input("Enter age:"))#taking value from user and storing it in age variable
if age<18:#comparing value of age
    print("Not eligible for vote")#printing the prompt
if age>18:#comparing value of age
    print("Eligible for vote")#printing the prompt

Enter age: 19


Eligible for vote


### Output of program
Eligible for vote

# TO DO Task:
## Program that takes input the length of hairs and determines the person is male or female
### Step 1:real word scenario:
program to find that person is male or female
### Step 2:client requirement
simple program that finds that person is male or female
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:float
### instruction:
comparision
### processing:
length>6
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find that person is male or female
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5

In [3]:
#input
length=float(input("Enter length of hairs in inches:"))#taking input from user and saving it in length variable
if length>6:
    print("Person is female")
else:
    print("Person is male")

Enter length of hairs in inches: 5


Person is male


### Step 10
### output of program
Person is male

# TO DO Task:
## Program that take input the age of a person and clarifies it in different categories that are:child,teenager,adult,senior
### Step 1:real word scenario:
program to find the group category of a person
### Step 2:client requirement
simple program that finds the group category of a person
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:int
### instruction:
logical comparision
### processing:
age>=0 and age<=12

age>=13 and age<=19

age>=20 and age<=64

age>=65
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find that person falls in which category
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=28

In [4]:
#input
age=int(input("Enter age:"))#taking age as input from user and store it in age variable
if age>=13 and age<=19:
    print("Child")
elif age>=13 and age<=19:
    print("Teenager")
elif age>=20 and age<=64:
    print("Adult")
elif age>=65:
    print("Senior")

Enter age: 28


Adult


### Output of program
Adult

# TO DO Task:
## Program that takes five number as an input from the user and prints the largest number
### Step 1:real word scenario:
program to find the largest number from five numbers
### Step 2:client requirement
simple program that finds the larget number
### Step 3:input-processing-output
### input:
No of inputs=5
Data types:(int,int,int,int,int)
### instruction:
comparision
### processing:
n1>n2

n2>n3

n3>n4

n4>n5
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find which number is largest
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
n1=3

n2=1

n3=9

n4=6

n5=0

In [6]:
#input
n1=int(input("Enter first number:"))#Taking first number
n2=int(input("Enter second number:"))#Taking second number
n3=int(input("Enter third number:"))#Taking third number
n4=int(input("Enter fourth number:"))#Taking fourth number
n5=int(input("Enter fifth number:"))#Taking fifth number
largest=n1#setting largest to n1
if n2>largest:#comparing largest and n2
    largest=n2
if n3>largest:#comparing largest and n3
    largest=n3
if n4>largest:#comparing largest and n4
    largest=n4
if n5>largest:#comparing largest and n5
    largest=n5
print("Largest number is:",largest)

Enter first number: 3
Enter second number: 1
Enter third number: 9
Enter fourth number: 6
Enter fifth number: 0


Largest number is: 9


### Step 10:
### Output of program
Largest number is:9

# TO DO Task:
## Program that take three sides of triangle as an input and tell whether the triangle is equilateral issosceles and scalene
### Step 1:real word scenario:
program to find the type of a triqangle
### Step 2:client requirement
simple program that finds the type of az triangle
### Step 3:input-processing-output
### input:
No of inputs=3
Data types:(int,int,int)
### instruction:
comparision
### processing:
side1 == side2 == side3

side1 == side2 or side1 == side3 or side2 == side3
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find the type of a triangle
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
side1=5

side2=5

side3=8

In [7]:
#input
side1=int(input("Enter side1 of a triangle:"))
side2=int(input("Enter side2 of a triangle:"))
side3=int(input("Enter side3 of a triangle:"))
if side1 == side2 == side3:
    print("The triangle is equilateral.")
elif side1 == side2 or side1 == side3 or side2 == side3:
    print("The triangle is isosceles.")
else:
    print("The triangle is scalene.")

Enter side1 of a triangle: 5
Enter side2 of a triangle: 5
Enter side3 of a triangle: 8


The triangle is isosceles.


### Step 10
### Output of a program
The triangle is isosceles

# TO DO Task:
## Program to find leap year
### Step 1:real word scenario:
program to find that year is leap or not
### Step 2:client requirement
simple program that finds the year is leap or not
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:int
### instruction:
mod
### processing:
year%400

year%100

year%4
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find that year is leap or not
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
2024

In [2]:
#input
year=int(input("Enter the year:"))#taking year as an input
if(year%400==0):#checking condition
    print("It is a leap year")
else:
    if(year%100==0):#checking condition
        print("It is not a leap year")
    else:
        if(year%4==0):#checking condition
            print("It is a leap year")
        else:
            print("It is not a leap year")

Enter the year: 2024


It is a leap year


### Step 10:
### Output of program
It is a leap year

# Your Turn:
## Program to find the grade
### Step 1:real word scenario:
program to find the grade
### Step 2:client requirement
simple program that finds grade
### Step 3:input-processing-output
### input:
No of inputs=1
Data types:float
### instruction:
comparision
### processing:
number>=90

number>=80

number>=70

number>=60
### Output:
No of output:1
Data type:string
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:To find the grade
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
75

In [3]:
#input
number=float(input("Enter your marks:"))
if number>=90:
    grade="A"
elif number>=80:
    grade="B"
elif number>=70:
    grade="C"
elif number>=60:
    grade="D"
else:
    grade="F"
print("Your grade is:",grade)

Enter your marks: 75


Your grade is: C


### Step 10
### Output of the program
Your grade is:C

# TO DO Task:
## Program that prints first 50 even numbers on output screen
### Step 1:real word scenario:
program that prints first 50 even numbers on screen
### Step 2:client requirement
simple program that prints first 50 even numbers
### Step 3:input-processing-output
### input:
No of inputs=0
### instruction:
comparision
### processing:
count<50
### Output:
No of output:50
Data type:int
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:That prints first 50 even numbers
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
no data needed

In [4]:
count=0
number=2
while count < 50:
    print(number)
    number+=2
    count+=1

2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


### Step 10
### output of program
2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100

# TO DO Task:
## Program that take a number from the user as an input and prints its table on the screen
### Step 1:real word scenario:
program that prints the table of a number on the screen
### Step 2:client requirement
simple program that prints the table of a number on the screen
### Step 3:input-processing-output
### input:
No of inputs=1
Data type of input:int
### instruction:
multiply
### processing:
number*i
### Output:
No of output:10
Data type:int
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:That prints the table of a number
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5

In [3]:
number=int(input("Enter a number: "))
for i in range(1, 11):
    print(f"{number} * {i} = {number * i}")

Enter a number:  5


5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


### Step 10
### Output of the program
Enter a number:  5
5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50

### TO DO Task:
### Program that take input from the user the number,lower bound, and upper bound and prints the table of the number from lower bound to the upper bound
### Step 1:real word scenario:
program that prints the table from lower bound till upper bound
### Step 2:client requirement
simple program that prints table from lower bound till upper bound
### Step 3:input-processing-output
### input:
No of inputs=3


Data type:int
### instruction:
multiplication
### processing:
number*i
### Output:
No of output:depends on lower and upper bound


Data type:int
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:That prints the table of a number from lower bound till upper bound
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
number=5


lower_bound=4


upper_bound=7

In [2]:
number=int(input("Enter a number: "))
lower_bound=int(input("Enter lower bound: "))
upper_bound=int(input("Enter upper bound: "))
for i in range(lower_bound, upper_bound + 1):
    print(f"{number} * {i} = {number * i}")

Enter a number:  5
Enter lower bound:  4
Enter upper bound:  7


5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35


# TO DO Task:
## Program that is user driven and that perform task that we have done previously
### Step 1:real word scenario:
program that is user driven
### Step 2:client requirement
simple program that is user driven
### Step 3:input-processing-output
### input:
No of inputs=depending upon the user what they choose
### instruction:
comparision,multiplication
### processing:
number*i,comparision
### Output:
No of output:depends on user what program they choose
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:user driven that give choices
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
choice=2

In [5]:
print("Press 1 to print table or press 2 to print first 50 even numbers or press 3 to exit the program")
choice=int(input("Enter your choice:"))
if choice==1:
    number=int(input("Enter a number: "))
    for i in range(1, 11):
        print(f"{number} * {i} = {number * i}")
if choice==2:
    count=0
    number=2
    while count < 50:
        print(number)
        number+=2
        count+=1
if choice==3:
    print("Program exit successfuly")

Press 1 to print table or press 2 to print first 50 even numbers or press 3 to exit the program


Enter your choice: 2


2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100


### Step 10
### Output of code
Press 1 to print table or press 2 to print first 50 even numbers or press 3 to exit the program


Enter your choice: 2


2
4
6
8
10
12
14
16
18
20
22
24
26
28
30
32
34
36
38
40
42
44
46
48
50
52
54
56
58
60
62
64
66
68
70
72
74
76
78
80
82
84
86
88
90
92
94
96
98
100

# TO DO Task:
program that prints following pattern:


1


12


123


1234


12345

### Step 1:real word scenario:
program that print pattern
### Step 2:client requirement
simple program that prints pattern
### Step 3:input-processing-output
### input:
No of inputs=0
### instruction:
printing of pattern
### processing:
printing using increment and decrement of variables
### Output:
No of output:5 lines
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:printing of a particular pattern
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
no data required

In [1]:
# Initialize the number of rows
n = 5
# Loop from 1 to n
for i in range(1, n + 1):
    for j in range(1, i + 1):
        print(j, end="")
    print()

1
12
123
1234
12345


### Step 10
### Output of program


1


12


123


1234


12345

# TO DO Task:
## Program that prints thw folowing pattern on the screen

### Step 1:real word scenario:
program that print pattern
### Step 2:client requirement
simple program that prints pattern
### Step 3:input-processing-output
### input:
No of inputs=0
### instruction:
printing of pattern
### processing:
printing using increment and decrement of variables
### Output:
No of output:5 lines
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:printing of a particular pattern
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
no data required

In [2]:
n = 5
for i in range(1, n + 1):
    for j in range(i):
        print('*', end='')
    print()

*
**
***
****
*****


### Step 10
### Output of the program

# TO DO Task:
## Program that prints the following pattern

### Step 1:real word scenario:
program that print pattern
### Step 2:client requirement
simple program that prints pattern
### Step 3:input-processing-output
### input:
No of inputs=0
### instruction:
printing of pattern
### processing:
printing using increment and decrement of variables
### Output:
No of output:5 lines
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:printing of a particular pattern
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
no data required

In [3]:
# Define the maximum number of stars in the middle row
max_stars = 5
# Loop to print the first half of the pattern
for i in range(1, max_stars + 1):
    print('*' * i)
# Loop to print the second half of the pattern
for i in range(max_stars - 1, 0, -1):
    print('*' * i)

*
**
***
****
*****
****
***
**
*


### Step 10
### Output of the program

# TO DO Task:
## Write a program that prints pyramid on the screen

### Step 1:real word scenario:
program that print pattern
### Step 2:client requirement
simple program that prints pattern
### Step 3:input-processing-output
### input:
No of inputs=0
### instruction:
printing of pattern
### processing:
printing using increment and decrement of variables
### Output:
pyramid on the screen
### Step 4: Pseudocode

### Step 5:Flowchart drawn
### Step 6: Suitable programming language,machine,ide
Programming language:Python

IDE:jupyter notebook

Operating system:windows 11

machine:i7
### Step 7:Developer information
Programmer name:Usfa Sohail

Id:usfasohail389@gmail.com

machine:i7

program name:printing of a pyramid pattern
### Step 8+9:Convert pseudocode code into code and test that code
### Data for testing the code:
no data required

In [4]:
# Number of rows for the pyramid
num_rows = 5
# Loop to print each row
for i in range(1, num_rows + 1):
    # Print leading spaces
    for _ in range(num_rows - i):
        print(' ', end='')
    # Print stars
    for _ in range(2 * i - 1):
        print('*', end='')
    # Move to the next line
    print()

    *
   ***
  *****
 *******
*********


## Step 10
### Output of program